### ISSUE TITLE
Web3.utils.hexToNumber wrong result

### ISSUE THREAD

**Issue by ndtuan89** [ethereum/web3.js#1677]
When I try to convert hex to number at the value

`Web3.utils.hexToNumber("0000000000000000000000000000000000000000000000000000000000000010");`

I got:

`10`

It should be 16 for the correct convert.

**Comment by nivida** [comment]
Depends on: https://github.com/SilentCicero/number-to-bn 

I'm in contact with Nick Dodson.

**Comment by ripern** [comment]
This is still broken. https://github.com/SilentCicero/number-to-bn has not changed in 2 years. Any alternative?

**Comment by nivida** [comment]
Will check if there is a similar module which is maintained. Otherwise, I will implement it on my own. @ripern

**Comment by nivida** [comment]
This got already fixed in v1.2.2.

Example:
``` javascript
console.log(utils.hexToNumber("0x0000000000000000000000000000000000000000000000000000000000000010"));
> 16
```
